ConfigurationSystem/Agent/Bdii2CSAgent.py

```diff
@@ -133,24 +133,30 @@ def __lookForNewCEs(self):
     """
 
     bannedCEs = self.am_getOption('BannedCEs', [])
-    res = getSiteCEMapping()
-    if not res['OK']:
-      return res
-    knownCEs = set()
-    for site in res['Value']:
-      knownCEs = knownCEs.union(set(res['Value'][site]))
-
-    knownCEs = knownCEs.union(set(bannedCEs))
 
     for vo in self.voName:
+      # get the known CEs for a given VO, so we can know the unknowns, or no longer supported,
+      # for a VO
+      res = getQueues(community=vo)
+      if not res['OK']:
+        return res
+
+      knownCEs = set()
+      for _site, ces in res['Value'].items():
+        knownCEs.union(ces.keys())
```

Comment:
```suggestion
        knownCEs.union(list(ces))
```

Reply:
Thanks for pointing out this line! Checking if casting is needed at all made me look at this logic.
The line itself at the moment is a no-op, as union doesn't update `knownCEs` but returns a new set. What I wanted was maybe `update`, but I need to check this logic again, it seems to work, which would mean this isn't necessary at all, but maybe I just added to much to bannedCEs in my instance.

(The casting is also not necessary at all `knownCEs.union(ces)` does it also)

Reply:
This should be, e.g., ``knownCEs.update(ces)``